File: app/src/main/java/com/example/pcmall/dialog/MessageDialog.java
package com.example.pcmall.dialog;

import android.content.Context;
import android.view.View;

import cn.pedant.SweetAlert.SweetAlertDialog;

public class MessageDialog extends SweetAlertDialog {

    public MessageDialog(Context context, int alertType) {
        super(context, alertType);
    }

    public void show() {
        super.show();
        super.getButton(SweetAlertDialog.BUTTON_CONFIRM).setVisibility(View.GONE);
        new Thread(() -> {
            try {
                Thread.sleep(1500);
                super.dismissWithAnimation();
            } catch (InterruptedException e) {
                throw new RuntimeException(e);
            }
        }).start();
    }
}

File: app/src/main/java/com/example/pcmallcompose/ui/dialog/MessageDialog.kt
package com.example.pcmallcompose.ui.dialog

import android.content.Context
import android.content.DialogInterface
import android.view.View
import cn.pedant.SweetAlert.SweetAlertDialog
import kotlinx.coroutines.DelicateCoroutinesApi
import kotlinx.coroutines.GlobalScope
import kotlinx.coroutines.delay
import kotlinx.coroutines.launch

class MessageDialog(
    context: Context,
    alertType: Int,
    title: String,
    dismissListener: ((DialogInterface) -> Unit)? = null
) : SweetAlertDialog(context, alertType) {

    init {
        super.setTitleText(title)
        super.setOnDismissListener(dismissListener)
    }

    @OptIn(DelicateCoroutinesApi::class)
    override fun show() {
        super.show()
        super.getButton(BUTTON_CONFIRM).visibility = View.GONE
        GlobalScope.launch {
            delay(1500)
            super.dismissWithAnimation()
        }
    }
}